server/test/unit/server/db/model/test_dispatch.py

```diff
@@ -614,6 +614,28 @@ def test_returns_dict(self):
 
         self.assertTrue(isinstance(call.as_dict(), dict))
 
+    def test_mongoengine_objects_serialized(self):
+        """
+        Ensure that mongoengine objects in the scheduled call were serialized.
+        """
+        schedule = bson.SON(SCHEDULE)
+        call = ScheduledCall.from_db(schedule)
+        call['principal'] = model.User('test_user')
+
+        result = call.as_dict()
+        self.assertTrue(isinstance(result['principal'], dict))
```

Comment:
This appears to only be checking that `result['principal']` is a dict, and possibly relying on `as_dict()` to throw exceptions if serialization fails. I don't see that anything is doing what I'd expect, based on the test name and docstring, which is making sure SCHEDULE is serialized properly, with all the expected keys, values, etc.